nexus/external-api/src/lib.rs

```diff
@@ -1281,12 +1283,41 @@ pub trait NexusExternalApi {
     ) -> Result<HttpResponseAccepted<views::FloatingIp>, HttpError>;
 
     // Multicast Groups
+    //
+    // API versioning note: Versioned endpoints can use default trait
+    // implementations when path types are identical between versions.
+    // `TypedBody<T>` has `.map()` for input conversion, but `Path<T>` does not.
+    // Endpoints with different path types (e.g., `v2025120300::MulticastGroupPath`
+    // vs `params::MulticastGroupPath`) must have implementations in
+    // `http_entrypoints.rs`.
+    //
+    // TODO: Consider adding `.map()` to dropshot's `Path<T>` (similar to
+    // `TypedBody<T>`) to enable default implementations for versioned endpoints
+    // with different path types.
+
+    /// List multicast groups.
+    #[endpoint {
+        method = GET,
+        path = "/v1/multicast-groups",
+        tags = ["experimental"],
+        operation_id = "multicast_group_list",
+        versions = ..VERSION_MULTICAST_IMPLICIT_LIFECYCLE_UPDATES,
+    }]
+    async fn v2025120300_multicast_group_list(
```

Comment:
Why does this need to exist? AFAICT it's identical in signature and implementation to `multicast_group_list`?

Reply:
The return types differ: v2025120300::MulticastGroup vs views::MulticastGroup. The default implementation converts the new type to the old type for backward compatibility with pre-implicit-lifecycle clients.